### PR TITLE
Pass proxy credentials to PowerShell

### PR DESCRIPTION
**Pull request by fgreinacher**
Fixes the problem mentioned in  https://github.com/microsoft/vscode-ripgrep/issues/5#issuecomment-811901359

### CODE REVIEW

**Comment by fgreinacher** [comment]
@slhultgren Would be great if you could check whether this solves the issue in your setup.

File: lib/download.js [hunk 1]
```diff
@@ -11,6 +11,7 @@ const child_process = require('child_process');
 const proxy_from_env = require('proxy-from-env');
 
 const packageVersion = require('../package.json').version;
+const { URL } = require('node:url');
```
Comment (roblourens) [inline lib/download.js:14]: What is this? Have never seen this syntax before

Comment (fgreinacher) [reply lib/download.js]: No idea, seems VS Code added it automatically. I removed it with 631ffc7fed9f9b3482386344274b2433b5c42e40.

**Comment by slhultgren** [comment]
@fgreinacher Thanks, this works well if the HTTP_PROXY env var includes user and password as unescaped text. 

For us we also do some basic "obfuscating" on the password so that it is at least not immediately human readable to %-- sequences, so I also had to add a line:
const decodedPassword = decodeURIComponent(password);

and use the decodedPassword for this part: "ConvertTo-SecureString '${decodedPassword}' -AsPlainText -Force".

**Comment by fgreinacher** [comment]
> @fgreinacher Thanks, this works well if the HTTP_PROXY env var includes user and password as unescaped text.
> 
> For us we also do some basic "obfuscating" on the password so that it is at least not immediately human readable to %-- sequences, so I also had to add a line:
> const decodedPassword = decodeURIComponent(password);
> 
> and use the decodedPassword for this part: "ConvertTo-SecureString '${decodedPassword}' -AsPlainText -Force".

Thanks! Updated with 45ac9ff599d1c3523a77d2c341f02d481ac36d77.

**Review by roblourens — APPROVED** [review]
Thanks!

**Pull request merged**